pkg/grpc/api.go

```diff
@@ -424,13 +426,38 @@ func (a *api) PublishEvent(ctx context.Context, in *runtimev1pb.PublishEventRequ
 
 	data := body
 
+	var cloudEventInterface interface{}
+	var publishTime string
+
+	if in.DataContentType == customCloudEvent {
+		unmarshalError := json.Unmarshal(body, &cloudEventInterface)
```

Comment:
Why the extra unmarshal? See how it is done for other attributes - I don't think we need this extra step.

/cc @yaron2

Reply:
Fixed !